wordpress-mobile/WordPress-iOS
# Remove extract notifications feature flag

Fixes #9877 

This PR removes all the extra code left from the `Extract Notification Interactions` project, plus the feature flag. In general it was a clean removal of "duplicated" code. (Code that's now implemented by `FormattableContent`.

The `ActionableObject` protocol was meant to be a temporary helper while `NotificationBlock` was still around, but now it also was removed. 

There was a small complication since `ActionContext` was depending on `ActionableObject`, and it was quite `Notification` specific. This issue was solved by making `ActionContext` generic on the block type it holds.

**To test:**
- Build and run.
- Check that `Notifications` displays and behaves as always.
- Check that Activity log events display and behave as they should.

*Note*:
Do not merge yet, until the destino is made to remove the feature flag.
